### ISSUE TITLE
Handling of .alpha in from-comfy converters

### ISSUE THREAD

**Issue by dxqb** [huggingface/diffusers#14088]
### Describe the bug

This is a question and maybe a bug report:
LoRAs in diffusers/peft-format don't support an `alpha` scaling factor. when converting another format to diffusers/peft, .alpha can be folded into lora_B]

kohya format LoRAs do hgave .alpha and for those models that diffusers implement a from-kohya converter, .alpha is supported.

For recent models, diffusers seems to  have switched to supporting comfy format rather than kohya - which is mostly: diffusion_model prefix, original model keys.
And there is the issue: some from-comfy converters in diffusers have .alpha supports, others don't.

The table below is AI generated from static analysis so I cannot guarantee its correctness, but I tripped over this for Anima comfy-format LoRAs, so that one I can confirm.

And here it becomes a question: are these converters supposed to support .alpha, or do you consider comfy-format to be without alpha?
Con: comfy-format isn't really defined. it's just whatever they support, and that varies wildly between models. They do support .alpha for all models though.
Pro: it's in use. For example, the kohya-to-comfy converter that ships with kohya-ss does output .alpha

| Model | kohya import | kohya alpha | comfy import | comfy alpha |
  |---|:--:|:--:|:--:|:--:|
  | StableDiffusion | ✅ | ✅ | ❌ | — |
  | StableDiffusionXL | ✅ | ✅ | ❌ | — |
  | Flux | ✅ | ✅ | ❌ | — |
  | Flux2 | ✅ | ✅ | ✅ | ❌ |
  | Wan | ✅ | ✅ | ✅ | ✅ |
  | Helios | ✅ | ✅ | ✅ | ✅ |
  | SkyReelsV2 | ✅ | ✅ | ✅ | ✅ |
  | QwenImage | ❌ | — | ✅ | ✅ |
  | ZImage | ❌ | — | ✅ | ✅ |
  | Ideogram4 | ❌ | — | ✅ | ✅ |
  | Anima | ❌ | — | ✅ | ❌ |
  | HiDreamImage | ❌ | — | ✅ | ❌ |
  | HunyuanVideo | ❌ | — | ✅ | ❌ |
  | LTX2 | ❌ | — | ✅ | ❌ |
  | LTXVideo | ❌ | — | ✅ | ❌ |
  | Lumina2 | ❌ | — | ✅ | ❌ |
  | ErnieImage | ❌ | — | ✅ | ❌ |

### Reproduction

Theoretical reprodution:
Load a LoRA generated by kohya-ss, converted to comfy-format by their to-comfy converter script, load that LoRA into diffusers.

If you need actual reproduction code or a file let me know

### Logs

```shell

```

### System Info

eb0a900ca

### Who can help?

Questions on LoRA: @sayakpaul

**Comment by dxqb** [comment]
I think I framed this partially incorrectly above. You haven't switched from kohya to a comfy-only converter, but one that can read kohya and comfy theoretically. But kohya imports fail because they have .alpha.

This is why Qwen and Z-Image actually load in diffusers in both formats (compare to table above), but not other models. Actual tests not static analysis:
| Model | KOHYA → diffusers | COMFY → diffusers | diffusers notes |
|-------|:-----------------:|:-----------------:|-----------------|
| qwen           | ✅ | ✅ | 
  | z-image        | ✅ | ✅ | 
| flux2          | ⊘ | ⊘ | KOHYA: no Flux2 kohya path that fits; COMFY: alpha-less converter |
  | flux           | ✅ | ⊘ | COMFY: can't read ComfyUI key names |

**Comment by sayakpaul** [comment]
Since these checkpoints are community-contributed and we don’t have any control over them, we fix them a bit adhoc but nonetheless prioritize them heavily.

**Comment by dxqb** [comment]
> Since these checkpoints are community-contributed and we don’t have any control over them, we fix them a bit adhoc but nonetheless prioritize them heavily.

*should* the converter support .alpha though, it's just not implemented everywhere?
asking because diffusers/peft peft format should *not*. It's defined to be without .alpha

I think so, but I'd like to hear your input before submitting a PR or two

**Comment by asomoza** [comment]
@dxqb just take into account that some old models don't have some conversions because people didn't ask for them, it could be for different reasons, but at the end it is because the loras or the model weren't popular, I don't think we should add more code for the repo to support something that's not going to be used, that's one of the reasons that we also wait for them to appear first.

In my opinion, the alphas don't add anything to the loras that makes them good, in fact, most of the time I find them performing worse in ComfyUI than in diffusers, you should probably follow what we have in the codebase but lets wait for @sayakpaul feedback.

**Comment by dxqb** [comment]
> [@dxqb](https://github.com/dxqb) just take into account that some old models don't have some conversions because people didn't ask for them

understood. because I'm running systematic tests currently, now would be the chance to change that and make the converters more consistent - if you're interested in such a PR.

> In my opinion, the alphas don't add anything to the loras that makes them good, in fact, most of the time I find them performing worse in ComfyUI than in diffusers, you should probably follow what we have in the codebase but lets wait for [@sayakpaul](https://github.com/sayakpaul) feedback.

It's not about whether alphas are useful or not. The kohya and comfy formats *use* them - so loading of these file formats either fails or is incorrect. Example Anima (AI-generated text, but confirmed by tests):


  ## TL;DR
  - **kohya silently does nothing** — *not* because `.alpha` is ignored, but because Anima's
    diffusers loader has **no kohya converter at all**. The `lora_unet_*` keys never get renamed
    to `transformer.*`, so the prefix filter drops the *entire* state dict. The "all keys contain
    'lora'" guard passes only by accident (the literal `lora_unet_` prefix contains "lora"), so
    nothing raises.
  - **comfy fails because of `.alpha`** — correct guess. After conversion the alpha keys become
    `transformer.…alpha`, which lack the "lora" substring and trip the validity guard. The error
    message is just generic and never names alpha.

  ## The mechanism (`AnimaLoraLoaderMixin`, `lora_pipeline.py:5629`)

  `lora_state_dict` only knows **two** branches:

  ```python
  has_diffusion_model = any(k.startswith("diffusion_model.") for k in state_dict)
  if has_diffusion_model:
      state_dict = _convert_non_diffusers_anima_lora_to_diffusers(state_dict)
  ```

  i.e. `transformer.*` (pass-through) or `diffusion_model.*` (comfy → convert). There is **no**
  `is_kohya` branch like `Flux2LoraLoaderMixin` has (`lora_pipeline.py:5867`).

  ### diffusers
  Keys already `transformer.…lora_A/lora_B`, no alpha keys at all (the export pre-folds scale into
  the weights). Pass-through → 560 keys → all injected. ✅

  ### comfy
  `has_diffusion_model` is True, so the converter runs. It renames
  `diffusion_model.blocks.0.cross_attn.k_proj.alpha` →
  `transformer.transformer_blocks.0.attn2.to_k.alpha`, preserving the `.alpha` suffix verbatim
  (`lora_conversion_utils.py:2374` — alpha is neither folded nor stripped). Back in
  `load_lora_weights`:

  ```python
  is_correct_format = all("lora" in key for key in state_dict.keys())  # 280 alpha keys fail this
  ```

  → raises. Confirmed: **all 280** failing keys are exactly the `.alpha` keys; the 560 weight keys
  are fine. So comfy would otherwise convert cleanly — alpha is the *sole* blocker.

  ### kohya
  `has_diffusion_model` is False (prefix is `lora_unet_`, not `diffusion_model.`) → converter is
  skipped. The guard `all("lora" in key)` is True (every key contains
  `lora_unet_`/`lora_down`/`lora_up`), so no error. Then:

  ```python
  transformer_state_dict      = {k: v for k,v if k.startswith("transformer.")}        # 0 keys
  text_conditioner_state_dict = {k: v for k,v if k.startswith("text_conditioner.")}   # 0 keys
  ```

  Both empty → neither `load_lora_into_transformer` nor `load_lora_into_text_conditioner` is called
  → **zero modules injected, zero warnings**. The model generates as base.

  ## So, on "is `.alpha` silently ignored?"
  For **kohya**, alpha is irrelevant — the whole dict is dropped before alpha matters. For **comfy**,
  alpha is the opposite of silently-ignored: it's the thing that actively breaks the load. And note
  this loader passes `network_alphas=None` to `load_lora_adapter` everywhere — there's no path that
  consumes a separate alpha tensor, which is why the diffusers-format export pre-bakes the scale into
  the weights instead.

**Comment by asomoza** [comment]
not sure if I'm writing to a human, but I think I expressed myself wrong, what I meant is not just for anima or a specific model, what I'm saying, is that in my personal opinion, the way we convert the loras that have alphas is ok (bake them),, I don't see a quality loss and even in some cases I see them improve, note that I'm basing my opinion on my eyes and what I see, not in numerical data.

in my personal opinion alphas are good for training, leaving them as a separate knob at inference doesn't improve the results and we also have granular scales for that.

**Comment by dxqb** [comment]
> not sure if I'm writing to a human

yes. I always mark or mention my AI use, as I have above.

> the way we convert the loras that have alphas is ok (bake them)

I agree. That would be okay. And that is what happens in some converters. But it does *not* happen in the Anima converter and others. Files with .alpha fail to import instead.
See the `comfy` section above.

**Comment by asomoza** [comment]
oh, now I finally understand what you're asking, that's the same reason as we stated before, we are pretty much reactive (but fast with it) when someone has trouble loading a lora, the difference between some converters handling or not the alphas is just because someone tried to load a lora with it or not.

So in short, anima doesn't have it because no one has opened an issue with a lora that didn't load that had alphas. Hope this clears your question about it.

About doing this as a massive update for all the models, that's an answer I defer to @sayakpaul
